### DISCUSSION TITLE
Duplicating an AnimationTree results in an instance- why?

### DISCUSSION THREAD

**Discussion by LordMcMutton** [godotengine/godot-proposals#10345] [Editor]
Ho' there!

I just lost some progress because, unbeknownst to me, the behavior of an Animation Tree Node is unique among Nodes when being duplicated.

Normally when you duplicate a Node, it results in a separate copy of that Node, but when you duplicate an Animation Tree Node, the two Nodes... are the same node? Instanced, I suppose.

So the work I did in a new Animation Tree was actually undoing the work I had already done in a previous Animation Tree

Is this intended, or a bug? If it's the former, can someone explain the logic behind it?

**Comment by AThousandShips** [comment]
Are you talking about the animation attached to the `AnimationTree`? That's not duplicated as it is a resource (and is exactly how other nodes work, like `CollisionShape2D`), or are you talking about properties of `AnimationTree` itself?

**Reply by LordMcMutton** [reply]
I mean the Node itself- they share an... internal node setup, I suppose

I duplicate the Node, like so:
![image](https://github.com/user-attachments/assets/c3c65621-724f-40e7-ad22-fc2520217c34)

And both Nodes share the same tree:
![image](https://github.com/user-attachments/assets/a6c4decb-1bf9-41bf-b7db-1ecb1e7be84e)

So if I delete the CurrentAnimation node in the upper left there, it will be gone in both AnimationTree_WalkRun and AnimationTree_Sprint

**Reply by AThousandShips** [reply]
They do not, that's a resource, the root node of the animation is a resource, [`tree_root`](https://docs.godotengine.org/en/latest/classes/class_animationtree.html#class-animationtree-property-tree-root), just make that unique and they won't be shared any longer

**Reply by LordMcMutton** [reply]
Thanks for explaining it! So it's a strange bit of structuring.

Not a fan of that, personally, as there's no indication that working with the Animation Tree Node isn't actually working with the 'real' part of the object, as it were- no clarity.

I'd say that's mostly due to the Resource system allowing individual Resources to be actual files or built-in (is the nomenclature Local and Remote?)

**Reply by AThousandShips** [reply]
It's not because of that, it's because of how resources are designed generally (local and remote is completely unrelated and has to do with debugging)

But this is a well established pattern and is used with most nodes that have complex data, like:
* CollisionShape2D/3D
* World environment
* TileMap(Layer)
* AnimatedSprite2D/3D

Having *this* complex data *not* be shared is extremely unusual, one case would be MultiplayerSynchronizer and MultiplayerSpawner, pretty much everything else uses resources for the configuration data

You also have to add the root node manually to your tree so you see that you add it and that it's a resource

**Reply by LordMcMutton** [reply]
To me it read more like picking what node layout you wanted to start with in this instance- I had no idea I was creating a Resource, which is why I encountered the problem.

The way Resources are designed is what I was talking about- how they can either be saved locally or not. I wasn't sure if there was an actual set of terms for either case.

**Reply by AThousandShips** [reply]
The term would be built-in and not, but the menu is the standard create resource menu, but you might just not be familiar with Godot's systems

But in any case the simple solution is to just make the resource unique, you just right click it and press "Make Unique" in the drop-down menu (or "Make Unique (recursive)", that would make sub nodes unique as well, if you so desire)

**Reply by LordMcMutton** [reply]
Oh-ho, I almost got the term right initially

But yes, the issue is basically just a lack of familiarity.

Thanks for going over it!
